### PR TITLE
Rwc2017 robot smach states

### PR DESCRIPTION
**Pull request by Rayman**
Please test an inspect (just run world_model.py, or the amigo console)

### CODE REVIEW

File: robot_smach_states/src/robot_smach_states/utility.py [hunk 1]
```diff
@@ -108,6 +109,48 @@ def execute(self, userdata=None):
 
 # ----------------------------------------------------------------------------------------------------
 
+
+class WaitForTriggerTimeout(smach.State):
```
Comment (Rayman) [inline robot_smach_states/src/robot_smach_states/utility.py]: @reinzor, you should merge this with `WaitForTrigger`

File: robot_smach_states/src/robot_smach_states/world_model.py [hunk 2]
```diff
@@ -27,34 +28,42 @@ def look_at_segmentation_area(robot, entity, volume=None):
     # Determine the height of the head target
     # Start with a default
     pos = entity.pose.frame.p
-    look_at_point_z = 0.7
+    x = pos.x()
+    y = pos.y()
+    z = 0.7
```
Comment (Rayman) [inline robot_smach_states/src/robot_smach_states/world_model.py]: @jlunenburg, these x, y and z are overwritten later

File: robot_smach_states/src/robot_smach_states/world_model.py [hunk 3]
```diff
@@ -27,34 +28,42 @@ def look_at_segmentation_area(robot, entity, volume=None):
...
         try:
-            look_at_point_z = search_volume.min_corner.z()
+            x_obj = 0.5 * (search_volume.min_corner.x() + search_volume.max_corner.x())
+            y_obj = 0.5 * (search_volume.min_corner.y() + search_volume.max_corner.y())
+            z_obj = search_volume.min_corner.z()
+            lookat_pos_map = entity.pose.frame * kdl.Vector(x_obj, y_obj, z_obj)
+            x = lookat_pos_map.x()
+            y = lookat_pos_map.y()
+            z = lookat_pos_map.z()
         except Exception as e:
```
Comment (Rayman) [inline robot_smach_states/src/robot_smach_states/world_model.py]: Why a try except? The object should be consistent.

File: robot_smach_states/src/robot_smach_states/world_model.py [hunk 4]
```diff
@@ -27,34 +28,42 @@ def look_at_segmentation_area(robot, entity, volume=None):
...
+    z = 0.7
 
     # Check if we have areas: use these
     if volume in entity.volumes:
         search_volume = entity.volumes[volume]
         try:
-            look_at_point_z = search_volume.min_corner.z()
+            x_obj = 0.5 * (search_volume.min_corner.x() + search_volume.max_corner.x())
+            y_obj = 0.5 * (search_volume.min_corner.y() + search_volume.max_corner.y())
+            z_obj = search_volume.min_corner.z()
```
Comment (Rayman) [inline robot_smach_states/src/robot_smach_states/world_model.py]: Why not the average z? or the max? (i.e. top of the table)

File: robot_smach_states/src/robot_smach_states/world_model.py [hunk 5]
```diff
@@ -27,34 +28,42 @@ def look_at_segmentation_area(robot, entity, volume=None):
...
+            lookat_pos_map = entity.pose.frame * kdl.Vector(x_obj, y_obj, z_obj)
```
Comment (Rayman) [inline robot_smach_states/src/robot_smach_states/world_model.py]: Why transform the point to z, you can just pass it on to the look_at

File: robot_smach_states/src/robot_smach_states/utility.py (hunk 1 above)
Comment (Rayman) [reply robot_smach_states/src/robot_smach_states/utility.py]: Rein told me the timeout version has different behavior so its a new state.

File: robot_smach_states/src/robot_smach_states/navigation/find.py [hunk 6]
```diff
@@ -74,10 +78,10 @@ def __init__(self,
 
     def execute(self, userdata=None):
         ids_to_select_from = [e.id for e in self._candidate_entities_designator.resolve()]
```
Comment (jlunenburg) [inline robot_smach_states/src/robot_smach_states/navigation/find.py]: Turn back into debugs

File: robot_smach_states/src/robot_smach_states/world_model.py (hunk 2 above)
Comment (jlunenburg) [reply robot_smach_states/src/robot_smach_states/world_model.py]: volume might not be in entity.volumes

File: robot_smach_states/src/robot_smach_states/world_model.py (hunk 4 above)
Comment (jlunenburg) [reply robot_smach_states/src/robot_smach_states/world_model.py]: It's the bottom of the _volume_, not the _entity_.

**Comment by MatthijsBurgh** [comment]
@Rayman create an issue?

File: robot_smach_states/src/robot_smach_states/navigation/find.py [hunk 7]
```diff
@@ -34,11 +34,15 @@ def entities_from_description(robot, entity_description, list_of_entity_ids=None
     # TODO: hack because ed maintains all labels that were ever assigned to an entity in the .types field
     if entity_description['type'] == 'person':
         entities = [e for e in entities if e.is_a('possible_human')]
+
+        # Remove the segmented entities from the inspection
```
Comment (LoyVanBeek) [inline robot_smach_states/src/robot_smach_states/navigation/find.py]: Is this always needed?

File: robot_smach_states/examples/simple_navigate.py [hunk 8]
```diff
@@ -22,26 +22,31 @@
...
+    #                                         ds.EntityByIdDesignator(robot, id=room))
+    # nav_state.execute()
+
     #nav_state = NavigateToSymbolic(robot, {
     #                                        ds.EntityByIdDesignator(robot, id="couch"):"in_front_of"},
     #                                        ds.EntityByIdDesignator(robot, id="couch"))
     #nav_state.execute()
+
+    nav_state = NavigateToWaypoint(robot=robot, waypoint_designator=ds.EntityByIdDesignator(robot=robot,
+                                                                                         id="hsr_demo_waypoint"),
```
Comment (MatthijsBurgh) [inline robot_smach_states/examples/simple_navigate.py]: why hsr_demo_waypoint??? Plus, why the change from NavigateToSymbolic to NavigateToWaypoint?

File: robot_smach_states/src/robot_smach_states/navigation/find.py (hunk 7 above)
Comment (MatthijsBurgh) [reply robot_smach_states/src/robot_smach_states/navigation/find.py]: I would call this a RoboCup Hack

File: robot_smach_states/src/robot_smach_states/world_model.py (hunk 2 above)
Comment (MatthijsBurgh) [reply robot_smach_states/src/robot_smach_states/world_model.py]: Than this should be in the 'else'

File: robot_smach_states/examples/simple_navigate.py (hunk 8 above)
Comment (jlunenburg) [reply robot_smach_states/examples/simple_navigate.py]: This is just an example, why care ;)

File: robot_smach_states/src/robot_smach_states/world_model.py (hunk 2 above)
Comment (jlunenburg) [reply robot_smach_states/src/robot_smach_states/world_model.py]: Fixed in 0346baf9c7e833cf19707e6b4ebfbcf62e463dd8

**Comment by Rayman** [comment]
Tested on robot
